### PR TITLE
[Metricbeat] Add vpc metricset for aws module

### LINKED ISSUE #14854: [Metricbeat] AWS Module: VPC Metricset(s)
There are three namespaces that provide VPC-related metrics:
* `AWS/NATGateway`
* `AWS/TransitGateway`
* `AWS/VPN`

# Metricbeat Module / Dataset release checklist

This checklist is intended for Devs which create or update a module to make sure modules are consistent.

## Modules

For a metricset to go GA, the following criterias should be met:

* [ ] Supported versions are documented
* [ ] Supported operating systems are documented (if applicable)
* [ ] Integration tests exist
* [ ] System tests exist
* [ ] Automated checks that all fields are documented
* [ ] Documentation
* [ ] Fields follow [ECS](https://github.com/elastic/ecs) and [naming conventions](https://www.elastic.co/guide/en/beats/devguide/master/event-conventions.html)
* [ ] Dashboards exists (if applicable)
* [ ] Kibana Home Tutorial (if applicable)
  * Open PR against Kibana repo with tutorial. Examples can be found [here](https://github.com/elastic/kibana/tree/master/src/legacy/core_plugins/kibana/server/tutorials).

## Metricbeat module

* [ ] Example `data.json` exists and an automated way to generate it exists (`go test -data`)
* [ ] Test environment in Docker exist for integration tests

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
## What does this PR do?

This PR is to add support for collecting vpc related monitoring metrics from CloudWatch. This includes AWS/NATGateway, AWS/VPN and AWS/TransitGateway namespaces.

## Why is it important?

Users can use this vpc metricset to collect metrics related to their VPCs, for example: bytes received/sent through VPN tunnel, total number of concurrent active TCP connections through NAT gateway, and number of bytes sent/received from transit gateway.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [ ] generate data.json
- [ ] dashboard

## How to test this PR locally

Make sure the AWS account you use for testing has VPC related metrics first. 
Enable vpc metricset and then change aws.yml to:
```
- module: aws
  period: 1m
  credential_profile_name: elastic-test
  metricsets:
    - vpc
```

## Related issues
elastic/beats#14854

### CODE REVIEW

**Comment by kaiyan-sheng** [comment]
Question: there are three namespaces included in this metricset, should they be split into 3 separate metricsets? 
My opinion is they are all VPC related, we can keep them together. But the concern is, by looking at the output events, `metricset.name` is vpc. But actual metrics are `aws.natgateway.metrics.*`, `aws.transitgateway.metrics.*` and `aws.vpn.metrics.*`.

**Comment by mtojek** [comment]
> Question: there are three namespaces included in this metricset, should they be split into 3 separate metricsets?
> My opinion is they are all VPC related, we can keep them together. But the concern is, by looking at the output events, `metricset.name` is vpc. But actual metrics are `aws.natgateway.metrics.*`, `aws.transitgateway.metrics.*` and `aws.vpn.metrics.*`.

I looked into the AWS docs to see if there're mentioned under a common VPC umbrella or separately. Is it possible to add a common key to these metrics? e.g. `aws.vpc.vpn.metrics.*`, `aws.vpc.natgateway.metrics.*`

File: x-pack/metricbeat/module/aws/vpc/vpc_test.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package vpc
+
+import (
+	"os"
+
+	"github.com/elastic/beats/metricbeat/mb"
```
Comment (mtojek) [inline x-pack/metricbeat/module/aws/vpc/vpc_test.go]: Shouldn't that be together with other `beats` imports?

**Comment by kaiyan-sheng** [comment]
> I looked into the AWS docs to see if there're mentioned under a common VPC umbrella or separately. Is it possible to add a common key to these metrics? e.g. `aws.vpc.vpn.metrics.*`, `aws.vpc.natgateway.metrics.*`

Thanks for your review!! Here is the AWS doc that mentioned these three namespaces under VPC: https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/aws-services-cloudwatch-metrics.html 
Hmm since this metricset is using light module to create, vpn/natgateway/transitgateway are from the given namespace. Similarly, for ELBs, metrics are started with `aws.elb`, `aws.applicationelb` and `aws.networkelb`.

File: x-pack/metricbeat/module/aws/vpc/vpc_test.go (hunk 1 above)
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/aws/vpc/vpc_test.go]: The other two `beats` import are for registering input module and metricset 😄 So I left them separate with a comment.

**Comment by kaiyan-sheng** [comment]
This will be tested by https://github.com/elastic/beats/pull/16892.

**Pull request merged**